src/mechanics/input.rs

```diff
@@ -137,12 +172,85 @@ pub fn move_player(
             )
             .is_some()
             {
-                player_movement_broadcast.send(PlayerMovementActions::Bumping);
+                entity_movement_broadcast.send(PlayerMovementActions::Bumping);
+                commands.entity(entity).remove::<MovementIntent>();
                 return;
             }
         }
-        player_transform.translation = projected_position;
-        player_movement_broadcast.send(PlayerMovementActions::Walking);
+
+        entity_transform.translation = projected_position;
+        entity_movement_broadcast.send(PlayerMovementActions::Walking);
+        commands.entity(entity).remove::<MovementIntent>();
+    }
+}
+
+pub fn interact_entity(
+    input: Res<Input<KeyCode>>, 
+    tile_query: Query<&EntityInstance>,
+    entity_query: Query<(&Transform, &DirectionFacing)>,
+    level_dimension: Res<LevelDimensions>,
+)
+{
+    if entity_query.is_empty() {
+        return;
+    }
+
+    if input.just_pressed(KeyCode::E) {
+        let interactive_tiles = tile_query
+            .iter()
+            .filter(|&tile| !tile.field_instances.is_empty())
+            .filter(|&tile| {
+                tile.field_instances
+                    .iter()
+                    .any(|field_instance| field_instance.identifier == "Interactable")
+            })
+            .collect::<Vec<&EntityInstance>>();
+
+        let (entity_transform, facing) = entity_query.get_single().unwrap();
```

Comment:
This will become a problem the moment we have multiple entities beyond the player, causing a crash. Since interaction only makes sense so far in the context of the player, I would recommend limiting the entity query to the player:

```rust
...
entity_query: Query<(&Transform, &DirectionFacing), With<Player>),
...
```

Reply:
Yikes, thanks for catching that